### ISSUE TITLE
Add the possibility to change the model (other than OpenAI-models)

### ISSUE THREAD

**Issue by mmuyakwa** [codecompanion-ai/code-companion#64]
You provide the ability to set the base_url.
`Groq` or other API's like `OpenRouter` can use the OpenAI API.
But they use different models.
The only thing missing is the ability to manually change the model to 
e.g. "mixtral-8x7b-32768" (via Groq) or perhaps "openai/gpt-4o" (via OpenRouter).

Since OpenAI changed its policy to not allow prepaid credit cards, many users in Europe lost the ability to use the OpenAI API directly.

**Comment by codecompanion-ai** [comment]
> You provide the ability to set the base_url. `Groq` or other API's like `OpenRouter` can use the OpenAI API. But they use different models. The only thing missing is the ability to manually change the model to e.g. "mixtral-8x7b-32768" (via Groq) or perhaps "openai/gpt-4o" (via OpenRouter).
> 
> Since OpenAI changed its policy to not allow prepaid credit cards, many users in Europe lost the ability to use the OpenAI API directly.

Makes sense. Other models are just not very good at coding, app might not be very useful or work at all. 
Making it flexible to use "custom" model makes sense anyways. I will add that. 

Another option, I might add paid version if you are interested with more flexible payment options that will not require you to use OpenAI/Anthropic API. Let me know if you will be interested in that or not. Just want to make it easy for app users

**Comment by mmuyakwa** [comment]
In essence, I would be willing to pay for the application. (NO SUBSCRIPTION! Payment per major version or preferably a lifetime subscription with a one-time payment.)

Since the context window is often not sufficient, an RAG extension would be helpful. It is annoying when the AI gets stuck in a loop and keeps trying approaches that previously did not work. It would be helpful if the AI could remember, based on RAG, what has been tested and whether it was successful or not.

**Comment by FellowTraveler** [comment]
Should be able to configure endpoints, models, sampler settings, embeddings of choice, etc. Should not be reliant on OpenAI for embeddings for example.
I would be happy to pay for this app in the app store, but I also want ability to experiment with whatever models I want.

**Comment by TeemuSo** [comment]
I like this request. It would be great to have freedom in the configurations, such that we could hook up router like https://www.notdiamond.ai/

**Comment by codecompanion-ai** [comment]
> Should be able to configure endpoints, models, sampler settings, embeddings of choice, etc. Should not be reliant on OpenAI for embeddings for example. I would be happy to pay for this app in the app store, but I also want ability to experiment with whatever models I want.

Made some changes on this. 

Now you have more freedom of picking Base model, Small model, Embeddings
you can add Base url
or add "custom" model name if not listed
and now we have integration with OpenRouter that has lots of different models
